snippets/puppet_setup.erb

```diff
@@ -32,7 +34,15 @@ cat > <%= etc_path %>/puppet.conf << EOF
 <%= snippet 'puppet.conf' %>
 EOF
 
-<% unless @host.param_true?('enable-puppetlabs-pc1-repo') -%>
+<% if @host.param_true?('enable-puppetlabs-pc1-repo') -%>
+<% if rhel_compatible -%>
```

Comment:
You mean change to `if @host.operatingsystem.family == 'Redhat'` and rely on `os_major > 6` doing the right thing for Fedora?

I was wary about trying to be too clever and coming up with correct, but difficult to follow logic.  I also wasn't in the position to test other OSes and didn't want to make assumptions.
I agree it's a bit messy though :(

Maybe it'd be best to refactor after other people contribute the correct and tested logic for some of the other OSes?  Dunno.